### PR TITLE
[web] improve types using a firstWhereOrNull utility

### PR DESCRIPTION
**Pull request by yjbanov**
Cleans up type casts in a couple of cases:

- Remove usage of `dynamic`.
- Remove the need to cast to nullable just to satisfy a limitation in `firstWhere`.

### CODE REVIEW

File: lib/web_ui/test/engine/semantics/semantics_tester.dart [hunk 1]
```diff
@@ -368,13 +368,13 @@ void expectSemanticsTree(String semanticsHtml) {
...
 /// Finds the first HTML element in the semantics tree used for scrolling.
 html.Element? findScrollable() {
-  return appHostNode.querySelectorAll('flt-semantics').cast<html.Element?>().firstWhere(
-        (html.Element? element) =>
-            element!.style.overflow == 'hidden' ||
-            element.style.overflowY == 'scroll' ||
-            element.style.overflowX == 'scroll',
-        orElse: () => null,
-      );
+  return appHostNode.querySelectorAll('flt-semantics').cast<html.Element>().firstWhereOrNull(
```
Comment (kevmoo) [inline lib/web_ui/test/engine/semantics/semantics_tester.dart:371]: you can drop the `cast`, too!

Comment (yjbanov) [reply lib/web_ui/test/engine/semantics/semantics_tester.dart:371]: I can't because `querySelectorAll` returns `Node`, not `Element.`

Comment (yjbanov) [reply lib/web_ui/test/engine/semantics/semantics_tester.dart:371]: Hold on a sec, that's a bug in our `HostNode` class. Stand by!

Comment (kevmoo) [reply lib/web_ui/test/engine/semantics/semantics_tester.dart:371]: Oh! On `Node`, not document. Ack.

Comment (yjbanov) [reply lib/web_ui/test/engine/semantics/semantics_tester.dart]: Done!

**Review by kevmoo — APPROVED** [review]

**Pull request merged**